### PR TITLE
ISTIO-SECURITY-2023-005 & CNI Docs Updates

### PR DESCRIPTION
**Pull request by pmerrison**
## Reviewers

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 1]
```diff
@@ -213,6 +213,14 @@ which detects if traffic redirection is set up correctly, and blocks the pod sta
 The CNI DaemonSet will detect and evict any pod stuck in such state. When the new pod starts up, it should have traffic redirection set up properly.
 This mitigation is enabled by default and can be turned off by setting `values.cni.repair.enabled` to false.
 
+This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/) .  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/cni/index.md:216]:
```suggestion
This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/).  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.
```

File: content/en/news/security/istio-security-2023-005/index.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+{{< security_bulletin >}}
+
+The Istio Security Committee were recently made aware of a potential scenario where the Istio CNI could be used as an attack vector on an already compromised node due to its high level of permissions.  The vector involves abusing the `istio-cni-repair-role` `ClusterRole` on a compromised node to expand the scope of the compromise from local to the node to a cluster-wide compromise.
+
+The Istio maintainers are, therefore, gradually rolling out a change to the above `ClusterRole` that reduces the permissions to close this potential attack vector. In the patched versions, roles are limited to the bare minimum requirements based on the [repair mode selected](/docs/setup/additional-setup/cni/#race-condition--mitigation). Previously, regardless of the configuration all roles were granted, and the roles that were granted were excessive.
+
+An additional option can further mitigate any potential attacks, by completely removing the need for Istio CNI to have custom RBAC permissions; due to the possible risks associated with this new method, it is only enabled by default on Istio 1.21+. See below for the configuration options available, and roles required:
```
Comment (ericvn) [inline content/en/news/security/istio-security-2023-005/index.md:20]:
```suggestion
An additional option can further mitigate any potential attacks, by completely removing the need for Istio CNI to have custom RBAC permissions. Due to the possible risks associated with this new method, it is only enabled by default on Istio 1.21+. See below for the configuration options available, and roles required:
```

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 3]
```diff
@@ -213,6 +213,14 @@ which detects if traffic redirection is set up correctly, and blocks the pod sta
 The CNI DaemonSet will detect and evict any pod stuck in such state. When the new pod starts up, it should have traffic redirection set up properly.
```
Comment (howardjohn) [inline content/en/docs/setup/additional-setup/cni/index.md:213]:
```suggestion
The CNI DaemonSet will detect and handle any pod stuck in such state; how the pod is handled is dependent on configuration described below.
```

(then see my next comment)

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 4]
```diff
@@ -213,6 +213,14 @@ which detects if traffic redirection is set up correctly, and blocks the pod sta
...
 This mitigation is enabled by default and can be turned off by setting `values.cni.repair.enabled` to false.
 
+This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/) .  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.
+
+|Configuration                    | Roles       | Notes
```
Comment (howardjohn) [inline content/en/docs/setup/additional-setup/cni/index.md:218]: Maybe we should have another column on how it works for each mode:

delete: pods are deleted, and when rescheduled they will have the correct configuration
label: pods are only labeled. This mode requires a user to detect when pods with these labels are set and take manual action (typically, deleting the pod)
repair: pods are dynamically reconfigured to have appropriate configuration. When the container restarts, the pod will continue normal execution.

File: content/en/news/security/istio-security-2023-005/index.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+|Configuration                    | Roles       | Notes
```
[the remark this answers is not in the document]
Comment (ericvn) [reply content/en/news/security/istio-security-2023-005/index.md:22]: Got it.

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 4 above)
Comment (ericvn) [reply content/en/docs/setup/additional-setup/cni/index.md]: @pmerrison Thoughts on John's comment.

File: content/en/news/security/istio-security-2023-005/index.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+title: ISTIO-SECURITY-2023-005
+subtitle: Security Bulletin
+description: Changes to Istio CNI RBAC permissions.
+cves: []
+cvss: N/A
+vector: N/A
+releases: ["All releases prior to 1.18.0", "1.18.0 to 1.18.5", "1.19.0 to 1.19.4", "1.20.0"]
+publishdate: 2023-12-11
```
Comment (dhawton) [inline content/en/news/security/istio-security-2023-005/index.md:9]:
```suggestion
publishdate: 2023-12-12
```

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 7]
```diff
@@ -210,9 +210,17 @@ The result is that the application pod comes up without Istio traffic redirectio
...
 This mitigation is enabled by default and can be turned off by setting `values.cni.repair.enabled` to false.
 
+This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/).  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.
+
+|Configuration                    | Roles       | Behavior on Error                                                                                                                           | Notes
+|---------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-------
+|`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration       .                                                           | Default in 1.20 and older
```
Comment (dhawton) [inline content/en/docs/setup/additional-setup/cni/index.md:220]:
```suggestion
|---------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-------
|`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration.                                                                  | Default in 1.20 and older
```

**Review by howardjohn — APPROVED** [review]

File: content/en/news/security/istio-security-2023-005/index.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+An additional option can further mitigate any potential attacks, by completely removing the need for Istio CNI to have custom RBAC permissions; due to the possible risks associated with this new method, it is only enabled by default on Istio 1.21+. See below for the configuration options available, and roles required:
+
+|Configuration                    | Roles       | Behavior on Error                                                                                                                           | Notes
+|---------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-------
+|`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration       .                                                           | Default in 1.20 and older
```
Comment (dhawton) [inline content/en/news/security/istio-security-2023-005/index.md:24]:
```suggestion
|`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration.                                                                  | Default in 1.20 and older
```

**Review by ericvn — APPROVED** [review]

File: content/en/news/security/istio-security-2023-005/index.md (hunk 8 above)
Comment (dhawton) [reply content/en/news/security/istio-security-2023-005/index.md:24]: The spaces here introduce a space between the last word and period https://deploy-preview-14302--preliminary-istio.netlify.app/latest/news/security/istio-security-2023-005/

**Comment by dhawton** [comment]
@pmerrison we good to drop the DNM label?

**Comment by ericvn** [comment]
I'm not sure if someone is coordinating all the patch releases. One comment I have to make is that the 1.20.1 release notes will require the manual cherry-pick (need data/args.yml update) which will add all the tests in the release-1.20 branch.

**Pull request merged**